src/Chartjs/Twig/ChartExtension.php

```diff
@@ -35,11 +43,13 @@ public function renderChart(Environment $env, Chart $chart, array $attributes =
     {
         $chart->setAttributes(array_merge($chart->getAttributes(), $attributes));
 
-        $html = '
-            <canvas
-                data-controller="'.trim($chart->getDataController().' symfony--ux-chartjs--chart').'"
-                data-view="'.twig_escape_filter($env, json_encode($chart->createView()), 'html_attr').'"
-        ';
+        $controllers = [];
+        if ($chart->getDataController()) {
+            $controllers[$chart->getDataController()] = [];
+        }
+        $controllers['symfony--ux-chartjs--chart'] = ['view' => $chart->createView()];
```

Comment:
I believe you can pass in the `symfony/ux-chartjs/chart` format here and Encore will convert it for you.